iodata/formats/sdf.py

```diff
@@ -0,0 +1,109 @@
+# IODATA is an input and output module for quantum chemistry.
+# Copyright (C) 2011-2019 The IODATA Development Team
+#
+# This file is part of IODATA.
+#
+# IODATA is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 3
+# of the License, or (at your option) any later version.
+#
+# IODATA is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, see <http://www.gnu.org/licenses/>
+# --
+"""SDF file format.
+
+Usually, the different frames in a trajectory describe different geometries of the same
+molecule, with atoms in the same order. The ``load_many`` and ``dump_many`` functions
+below can also handle an SDF file with different molecules, e.g. a molecular database.
+"""
+
+
+from typing import TextIO, Iterator
+
+import numpy as np
+
+from ..docstrings import (document_load_one, document_load_many, document_dump_one,
+                          document_dump_many)
+from ..iodata import IOData
+from ..periodic import sym2num, num2sym
+from ..utils import angstrom, LineIterator
+
+
+__all__ = []
+
+
+PATTERNS = ['*.sdf']
+
+
+@document_load_one("SDF", ['atcoords', 'atnums', 'title'])
+def load_one(lit: LineIterator) -> dict:
+    """Do not edit this docstring. It will be overwritten."""
+    title = next(lit).strip()
+    # The next two lines are general comments
+    next(lit)
+    next(lit)
+    words = next(lit).split()
+    size = int(words[0])
+    atcoords = np.empty((size, 3), float)
+    atnums = np.empty(size, int)
+    for i in range(size):
+        words = next(lit).split()
+        atcoords[i, 0] = float(words[0]) * angstrom
+        atcoords[i, 1] = float(words[1]) * angstrom
+        atcoords[i, 2] = float(words[2]) * angstrom
+        try:
```

Comment:
I assume this try-except block is based on the XYZ code, but the XYZ code is actually a poor example of how to use dictionaries in Python. Both should be fixed ideally, to avoid making this suggestion. In general in Python, if there is a way to get the same effect without try-except, then it is preferable for clarity (and speed). In this case, it could also be done like this:

```python
atnum = sym2num.get(words[3].title())
if atnum is None:
    atnum = int(words[3])
atnums[i] = atnum
```

Note that the fallback option should also have index 3. I'm not sure if SDF ever uses integers instead of symbols at index 3, so the fallback option may be redundant anyway? If the fallback is not needed, it could become one line instead:

```python
atnum = sym2num.get(words[3].title(), 0)
```

Reply:
Your are right, only elements are used. I changed it to your suggestion of only line.